TestStack.Dossier/DataSources/Picking/RandomItemSource.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace TestStack.Dossier.DataSources.Picking
+{
+    /// <summary>
+    /// Implements the random item strategy
+    /// </summary>
+    public class RandomItemSource<T> : DataSource<T>
+    {
+        /// <inheritdoc />
+        public RandomItemSource(IList<T> list)
+        {
+            Data = list;
+            Generator.StartIndex = 0;
+            Generator.ListSize = Data.Count;
+        }
+
+        /// <inheritdoc />
+        protected override IList<T> InitializeDataSource()
+        {
+            // This method will never be called as the list is set in the constructor.
+            throw new NotImplementedException();
```

Comment:
The thing is there is already a _list field. I don't really like the idea of a second field to store the same thing. 

One option is to make the field protected, in which case it becomes available to this constructor and I can just set it directly.  The reason I didn't go for that though is it then becomes accessible to *every* class that inherits from it, something I didn't really want to do. Making the Setter internal achieves what I want, because I can set it from this class but consumers don't gain access to the private field.

I could return new List<T>() from the InitializeDataSource method, but I thought that was misleading as it implied it might get called. This method is only called in the lazy instantiation of the list, meaning it will never be called in this implementation as we are setting the list in the constructor. If someone were to inherit from this class and call this method I'm happy it would give them an exception. They can just override it in their derived class with the behaviour they expect.

That was my thinking anyway. Let me know if you disagree at all. :-)